src/mixins/boundary.js

```diff
@@ -0,0 +1,60 @@
+import * as THREE from "three";
+
+export const BoundaryMixin = (superclass) => class extends superclass {
+
+    constructor(config) {
+        super(config);
+        this.boundaryConditions = config.boundaryConditions || {};
+    }
+
+    /**
+     * Boundaries are drawn only if type is "bc1", "bc2", or "bc3".
+     */
+    get areBoundariesEnabled() {return ["bc1", "bc2", "bc3"].includes(this.boundaryConditions.type)}
+
+    /**
+     * Returns a plane-like mesh object with given corner vertices in counterclockwise order.
+     */
+    getPlaneObject(color, coordinates1, coordinates2, coordinates3, coordinates4, zOffset = 0) {
+
+        const point1 = new THREE.Vector3(...coordinates1).add(new THREE.Vector3(0, 0, zOffset));
+        const point2 = new THREE.Vector3(...coordinates2).add(new THREE.Vector3(0, 0, zOffset));
+        const point3 = new THREE.Vector3(...coordinates3).add(new THREE.Vector3(0, 0, zOffset));
+        const point4 = new THREE.Vector3(...coordinates4).add(new THREE.Vector3(0, 0, zOffset));
+
+        const geometry = new THREE.Geometry();
+        geometry.vertices.push(point1, point2, point3, point4);
+        const face1 = new THREE.Face3(0, 1, 2);
+        const face2 = new THREE.Face3(2, 3, 0);
+        geometry.faces.push(face1, face2);
+        geometry.computeFaceNormals();
+        geometry.computeVertexNormals();
+
+        var material = new THREE.MeshBasicMaterial({
+            color: color,
+            opacity: 0.5,
+            transparent: true,
+            side: THREE.DoubleSide
+        });
+
+        return new THREE.Mesh(geometry, material);
+
+    }
+
+    /**
+     * Draw boundaries with +/- [L_z/2 + offset] z coordinates.
```

Comment:
1. offset => this.boundaryConditions.offset

Reply:
Done.